src/libxrpl/protocol/STIssue.cpp

```diff
@@ -65,18 +86,40 @@ STIssue::getSType() const
     return STI_ISSUE;
 }
 
+std::string
+STIssue::getText() const
+{
+    return asset_.getText();
+}
+
 Json::Value
 STIssue::getJson(JsonOptions) const
 {
-    return to_json(issue_);
+    Json::Value jv;
+    asset_.setJson(jv);
+    return jv;
 }
 
 void
 STIssue::add(Serializer& s) const
 {
-    s.addBitString(issue_.currency);
-    if (!isXRP(issue_.currency))
-        s.addBitString(issue_.account);
+    if (holds<Issue>())
+    {
+        s.addBitString(asset_.get<Issue>().currency);
+        if (!isXRP(asset_.get<Issue>().currency))
+            s.addBitString(asset_.get<Issue>().account);
+    }
+    else
+    {
+        s.addBitString(asset_.get<MPTIssue>().getIssuer());
+        s.addBitString(noAccount());
+        std::uint32_t sequence;
+        memcpy(
+            &sequence,
+            asset_.get<MPTIssue>().getMptID().data(),
+            sizeof(sequence));
+        s.add32(sequence);
```

Comment:
Let's not call `get` more than necessary

```suggestion
        auto const& issue = asset_.get<Issue>();
        s.addBitString(issue.currency);
        if (!isXRP(issue.currency))
            s.addBitString(issue.account);
    }
    else
    {
        auto const& issue = asset_.get<MPTIssue>();
        s.addBitString(issue.getIssuer());
        s.addBitString(noAccount());
        std::uint32_t sequence;
        memcpy(&sequence, issue.getMptID().data(), sizeof(sequence));
```